test/e2e/install_test.go

```diff
@@ -99,10 +109,137 @@ var _ = Describe("Operator Install", func() {
+		Expect(err).To(Not(HaveOccurred()))
+		err = ioutil.WriteFile(artifactPath+catalogsource.Name+"-catalogsource.yaml", catalogsourceYaml, 0644)
+		Expect(err).To(Not(HaveOccurred()))
+	}
+
+	// Get all Bundles in the namespace and save them to the artifact path.
+	bundles := rukpakv1alpha1.BundleList{}
+	err = c.List(ctx, &bundles, client.InNamespace(""))
+	Expect(err).To(Not(HaveOccurred()))
+	for _, bundle := range bundles.Items {
+		// Save bundle to artifact path
+		bundleYaml, err := yaml.Marshal(bundle)
+		Expect(err).To(Not(HaveOccurred()))
+		err = ioutil.WriteFile(artifactPath+bundle.Name+"-bundle.yaml", bundleYaml, 0644)
+		Expect(err).To(Not(HaveOccurred()))
+	}
+
+	// Get all BundleDeployments in the namespace and save them to the artifact path.
+	bundleDeployments := rukpakv1alpha1.BundleDeploymentList{}
+	err = c.List(ctx, &bundleDeployments, client.InNamespace(""))
+	Expect(err).To(Not(HaveOccurred()))
+	for _, bundleDeployment := range bundleDeployments.Items {
+		// Save bundleDeployment to artifact path
+		bundleDeploymentYaml, err := yaml.Marshal(bundleDeployment)
+		Expect(err).To(Not(HaveOccurred()))
+		err = ioutil.WriteFile(artifactPath+bundleDeployment.Name+"-bundleDeployment.yaml", bundleDeploymentYaml, 0644)
+		Expect(err).To(Not(HaveOccurred()))
+	}
+
+	for _, namespace := range namespaces.Items {
+		// let's ignore kube-* namespaces.
+		if strings.Contains(namespace.Name, "kube-") {
+			continue
+		}
+
+		namespaceArtifactPath := artifactPath + namespace.Name + "/"
+		err = os.Mkdir(namespaceArtifactPath, 0755)
+		Expect(err).To(Not(HaveOccurred()))
+
+		// get all deployments in the namespace and save them to the artifact path.
+		deployments, err := kubeClient.AppsV1().Deployments(namespace.Name).List(ctx, metav1.ListOptions{})
+		Expect(err).To(Not(HaveOccurred()))
+		for _, deployment := range deployments.Items {
+			// Save deployment to artifact path
+			deploymentYaml, err := yaml.Marshal(deployment)
+			Expect(err).To(Not(HaveOccurred()))
+			err = ioutil.WriteFile(namespaceArtifactPath+deployment.Name+"-deployment.yaml", deploymentYaml, 0644)
+			Expect(err).To(Not(HaveOccurred()))
+		}
+
+		// Get logs from all pods in all namespaces
+		pods, err := kubeClient.CoreV1().Pods(namespace.Name).List(ctx, metav1.ListOptions{})
+		Expect(err).To(Not(HaveOccurred()))
+		for _, pod := range pods.Items {
+			if pod.Status.Phase != v1.PodRunning {
+				continue
+			}
+			for _, container := range pod.Spec.Containers {
+				logs, err := kubeClient.CoreV1().Pods(namespace.Name).GetLogs(pod.Name, &v1.PodLogOptions{Container: container.Name}).Stream(ctx)
+				Expect(err).To(Not(HaveOccurred()))
+				defer logs.Close()
+
+				// Read logs
+				buf := make([]byte, 1024)
+				for {
+					_, err := logs.Read(buf)
+					if err != nil {
+						break
+					}
+				}
+
+				// Save logs to artifact path
+				err = ioutil.WriteFile(namespaceArtifactPath+pod.Name+"-"+container.Name+"-logs.txt", buf, 0644)
```

Comment:
To avoid reading the entire log in memory, then writing it to a file, I would recommend writing to the file as you read from the stream. Hopefully you can use io.Copy for this.

Reply:
Yes, you should be able to get a `Writer` to the output file and use `io.Copy()` with `logs` and the file `Writer`.

Reply:
fixed